packages/rich-text/src/test/get-active-object.js

```diff
@@ -3,13 +3,12 @@
  */
 
 import { getActiveObject } from '../get-active-object';
-import { OBJECT_REPLACEMENT_CHARACTER } from '../special-characters';
 
 describe( 'getActiveObject', () => {
 	it( 'should return object if selected', () => {
 		const record = {
 			replacements: [ { type: 'img' } ],
-			text: OBJECT_REPLACEMENT_CHARACTER,
+			text: ' ',
```

Comment:
is this being replaced with a space because we don't yet have the new RichText state value? I hope that eventually we can eliminate the character altogether, otherwise the space isn't all that different from the object replacement character, is it? a tradeoff between something that occasionally renders improperly and something that carries semantic meaning?

Reply:
I'm not sure I'm following here. The space has no semantic meaning outside of the rich text value, it's just plain text. When inside a rich text value, it has meaning because a replacement is stored at the same index.

How would we be able to eliminate the character? It's possible for an inline image to have formatting, so it needs to be possible to store formatting at the index.

Reply:
With #29870, it _could_ be possible to do it if we allow a range to start and end at the same index. 🤔

Reply:
Are there any formats beside a link that can apply to an inline image in a meaningful way?

Formatting to a single index doesn't seem too bad. This is how the WordPress.com bell notifications insert icons and images. at that point it doesn't require any in-band reference.

if we have to have formatting of inline images though, it means we have to first ensure we insert the image before applying the other formats.

since this is not a question really about this PR I don't want it to stall what's going on here. though, it does feel a little bit like what we're doing is trading tradeoffs here. in a buggy situation the object replacement character won't show (which is very good), but a space will be there, which it shouldn't, and there's no way to tell if that space was meant to be a space or a placeholder (which the object replacement character did convey)

Reply:
Why shouldn't there be a space? If there's a space, at that point it means you've converted HTML to plain text, so it's totally fine?